nav2_costmap_2d/include/nav2_costmap_2d/costmap_2d_publisher.hpp

```diff
@@ -151,8 +151,10 @@ class Costmap2DPublisher
   // Service for getting the costmaps
   rclcpp::Service<nav2_msgs::srv::GetCostmap>::SharedPtr costmap_service_;
 
-  nav_msgs::msg::OccupancyGrid grid_;
-  nav2_msgs::msg::Costmap costmap_raw_;
+  float grid_resolution;
```

Comment:
I'm not entirely clear why these were added. Can you explain these new member variables? Looking at the use, it seems like you can just use the information in their scopes.

Reply:
To store the previous state of grid i.e. resolution, width and height. 
If you look at [this](https://github.com/ros-planning/navigation2/blob/628bf78e75e733433c6500aa8ccea2574d65eb16/nav2_costmap_2d/src/costmap_2d_publisher.cpp#L184-L193) part of the code, grid resolution, width and height are being checked for in the if condition which is before the `prepareGrid()` call, where `grid_` is actually instantiated. The std::move in the publish call nulls out the `grid_` pointer, so I don't have the required information when this condition is checked the next time.

Reply:
Ok thats fine

Reply:
Just need merge conflicts and linters enabled and good to go